### ISSUE TITLE
Crash when trying to save theme

### ISSUE THREAD

**Issue by AnonWombat** [florisboard/florisboard#1643]
#### Environment information
- FlorisBoard Version: 0.3.14-rc01 (71)
- Device: OnePlus GM1917
- Android: 12 (cn=S sdk=31)

#### Features enabled
```kt
smartbar = true
suggestions = false
suggestions_clipboard = true
suggestions_next_word = true
glide = false
clipboard_internal = false
clipboard_history = false
```

#### Attached stacktrace files
<details>
<summary>1647223735481.stacktrace</summary>

```
java.lang.IllegalStateException: Check failed.
	at dev.patrickgold.florisboard.snygg.value.SnyggStringValueSpec.pack-RQoFFkw(SnyggValueSpec.kt:123)
	at dev.patrickgold.florisboard.snygg.value.SnyggFunctionValueSpec.pack-RQoFFkw(SnyggValueSpec.kt:144)
	at dev.patrickgold.florisboard.snygg.value.SnyggDefinedVarValue$Companion.serialize-IoAF18A(SnyggVarValue.kt:36)
	at dev.patrickgold.florisboard.snygg.SnyggStylesheetSerializer.serialize(SnyggStylesheet.kt:259)
	at dev.patrickgold.florisboard.snygg.SnyggStylesheetSerializer.serialize(SnyggStylesheet.kt:250)
	at kotlinx.serialization.json.internal.StreamingJsonEncoder.encodeSerializableValue(StreamingJsonEncoder.kt:219)
	at kotlinx.serialization.json.Json.encodeToString(Json.kt:85)
	at dev.patrickgold.florisboard.app.ui.ext.ExtensionEditScreenKt$EditScreen$1.invoke$handleSave(ExtensionEditScreen.kt:932)
	at dev.patrickgold.florisboard.app.ui.ext.ExtensionEditScreenKt$EditScreen$1.access$invoke$handleSave(ExtensionEditScreen.kt:226)
	at dev.patrickgold.florisboard.app.ui.ext.ExtensionEditScreenKt$EditScreen$1$2$1$2.invoke(ExtensionEditScreen.kt:315)
	at dev.patrickgold.florisboard.app.ui.ext.ExtensionEditScreenKt$EditScreen$1$2$1$2.invoke(ExtensionEditScreen.kt:314)
	at androidx.compose.foundation.ClickableKt$clickable$4$gesture$1$2.invoke-k-4lQ0M(Clickable.kt:153)
	at androidx.compose.foundation.ClickableKt$clickable$4$gesture$1$2.invoke(Clickable.kt:142)
	at androidx.compose.foundation.gestures.TapGestureDetectorKt$detectTapAndPress$2$1$1.invokeSuspend(TapGestureDetector.kt:223)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTaskKt.resume(DispatchedTask.kt:178)
	at kotlinx.coroutines.DispatchedTaskKt.dispatch(DispatchedTask.kt:166)
	at kotlinx.coroutines.CancellableContinuationImpl.dispatchResume(CancellableContinuationImpl.kt:397)
	at kotlinx.coroutines.CancellableContinuationImpl.resumeImpl(CancellableContinuationImpl.kt:431)
	at kotlinx.coroutines.CancellableContinuationImpl.resumeImpl$default(CancellableContinuationImpl.kt:420)
	at kotlinx.coroutines.CancellableContinuationImpl.resumeWith(CancellableContinuationImpl.kt:328)
	at androidx.compose.ui.input.pointer.SuspendingPointerInputFilter$PointerEventHandlerCoroutine.offerPointerEvent(SuspendingPointerInputFilter.kt:511)
	at androidx.compose.ui.input.pointer.SuspendingPointerInputFilter.dispatchPointerEvent(SuspendingPointerInputFilter.kt:406)
	at androidx.compose.ui.input.pointer.SuspendingPointerInputFilter.onPointerEvent-H0pRuoY(SuspendingPointerInputFilter.kt:419)
	at androidx.compose.ui.input.pointer.Node.dispatchMainEventPass(HitPathTracker.kt:310)
	at androidx.compose.ui.input.pointer.Node.dispatchMainEventPass(HitPathTracker.kt:297)
	at androidx.compose.ui.input.pointer.Node.dispatchMainEventPass(HitPathTracker.kt:297)
	at androidx.compose.ui.input.pointer.Node.dispatchMainEventPass(HitPathTracker.kt:297)
	at androidx.compose.ui.input.pointer.Node.dispatchMainEventPass(HitPathTracker.kt:297)
	at androidx.compose.ui.input.pointer.NodeParent.dispatchMainEventPass(HitPathTracker.kt:179)
	at androidx.compose.ui.input.pointer.HitPathTracker.dispatchChanges(HitPathTracker.kt:98)
	at androidx.compose.ui.input.pointer.PointerInputEventProcessor.process-BIzXfog(PointerInputEventProcessor.kt:80)
	at androidx.compose.ui.platform.AndroidComposeView.sendMotionEvent-8iAsVTc(AndroidComposeView.android.kt:1159)
	at androidx.compose.ui.platform.AndroidComposeView.handleMotionEvent-8iAsVTc(AndroidComposeView.android.kt:1109)
	at androidx.compose.ui.platform.AndroidComposeView.dispatchTouchEvent(AndroidComposeView.android.kt:1059)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3120)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2801)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3120)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2801)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3120)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2801)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3120)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2801)
	at com.android.internal.policy.DecorView.superDispatchTouchEvent(DecorView.java:502)
	at com.android.internal.policy.PhoneWindow.superDispatchTouchEvent(PhoneWindow.java:1890)
	at android.app.Activity.dispatchTouchEvent(Activity.java:4195)
	at com.android.internal.policy.DecorView.dispatchTouchEvent(DecorView.java:460)
	at android.view.View.dispatchPointerEvent(View.java:14799)
	at android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent(ViewRootImpl.java:6359)
	at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:6160)
	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:5638)
	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:5695)
	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:5661)
	at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:5826)
	at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:5669)
	at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:5883)
	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:5642)
	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:5695)
	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:5661)
	at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:5669)
	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:5642)
	at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:8582)
	at android.view.ViewRootImpl.doProcessInputEvents(ViewRootImpl.java:8533)
	at android.view.ViewRootImpl.enqueueInputEvent(ViewRootImpl.java:8502)
	at android.view.ViewRootImpl$WindowInputEventReceiver.onInputEvent(ViewRootImpl.java:8705)
	at android.view.InputEventReceiver.dispatchInputEvent(InputEventReceiver.java:259)
	at android.os.MessageQueue.nativePollOnce(Native Method)
	at android.os.MessageQueue.next(MessageQueue.java:335)
	at android.os.Looper.loopOnce(Looper.java:161)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7840)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:550)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)

```
</details>

<details>
<summary>1647220967388.stacktrace</summary>

```
java.lang.IllegalArgumentException: The percent should be in the range of [0, 100]
	at androidx.compose.foundation.shape.PercentCornerSize.<init>(CornerSize.kt:95)
	at androidx.compose.foundation.shape.CornerSizeKt.CornerSize(CornerSize.kt:82)
	at androidx.compose.foundation.shape.RoundedCornerShapeKt.RoundedCornerShape(RoundedCornerShape.kt:195)
	at dev.patrickgold.florisboard.app.ui.settings.theme.SnyggValueIconKt.alwaysPercentShape(SnyggValueIcon.kt:166)
	at dev.patrickgold.florisboard.app.ui.settings.theme.SnyggValueIconKt.SnyggValueIcon(SnyggValueIcon.kt:104)
	at dev.patrickgold.florisboard.app.ui.settings.theme.ThemeEditorScreenKt$ThemeEditorScreen$1$5$4$2$1$1$4.invoke(ThemeEditorScreen.kt:311)
	at dev.patrickgold.florisboard.app.ui.settings.theme.ThemeEditorScreenKt$ThemeEditorScreen$1$5$4$2$1$1$4.invoke(ThemeEditorScreen.kt:311)
	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:107)
	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:34)
	at androidx.compose.runtime.RecomposeScopeImpl.compose(RecomposeScopeImpl.kt:140)
	at androidx.compose.runtime.ComposerImpl.recomposeToGroupEnd(Composer.kt:2158)
	at androidx.compose.runtime.ComposerImpl.skipToGroupEnd(Composer.kt:2427)
	at androidx.compose.material.TextKt.ProvideTextStyle(Text.kt:266)
	at androidx.compose.material.ListItemKt$applyTextStyle$1$1.invoke(ListItem.kt:425)
	at androidx.compose.material.ListItemKt$applyTextStyle$1$1.invoke(ListItem.kt:424)
	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:107)
	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:34)
	at androidx.compose.runtime.CompositionLocalKt.CompositionLocalProvider(CompositionLocal.kt:228)
	at androidx.compose.material.ListItemKt$applyTextStyle$1.invoke(ListItem.kt:424)
	at androidx.compose.material.ListItemKt$applyTextStyle$1.invoke(ListItem.kt:423)
	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:107)
	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:34)
	at androidx.compose.material.TwoLine$ListItem$1$4.invoke(ListItem.kt:267)
	at androidx.compose.material.TwoLine$ListItem$1$4.invoke(ListItem.kt:261)
	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:107)
	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:34)
	at androidx.compose.runtime.RecomposeScopeImpl.compose(RecomposeScopeImpl.kt:140)
	at androidx.compose.runtime.ComposerImpl.recomposeToGroupEnd(Composer.kt:2158)
	at androidx.compose.runtime.ComposerImpl.skipToGroupEnd(Composer.kt:2427)
	at androidx.compose.material.ListItemKt.OffsetToBaselineOrCenter-Kz89ssw(ListItem.kt:415)
	at androidx.compose.material.ListItemKt.access$OffsetToBaselineOrCenter-Kz89ssw(ListItem.kt:1)
	at androidx.compose.material.TwoLine.ListItem(ListItem.kt:255)
	at androidx.compose.material.ListItemKt.ListItem(ListItem.kt:96)
	at dev.patrickgold.jetpref.material.ui.JetPrefListItemKt.JetPrefListItem(JetPrefListItem.kt:58)
	at dev.patrickgold.florisboard.app.ui.settings.theme.ThemeEditorScreenKt$ThemeEditorScreen$1$5$4$2$1.invoke(ThemeEditorScreen.kt:302)
	at dev.patrickgold.florisboard.app.ui.settings.theme.ThemeEditorScreenKt$ThemeEditorScreen$1$5$4$2$1.invoke(ThemeEditorScreen.kt:276)
	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:116)
	at androidx.compose.runtime.internal.ComposableLambdaImpl$invoke$1.invoke(ComposableLambda.jvm.kt:127)
	at androidx.compose.runtime.internal.ComposableLambdaImpl$invoke$1.invoke(ComposableLambda.jvm.kt:127)
	at androidx.compose.runtime.RecomposeScopeImpl.compose(RecomposeScopeImpl.kt:140)
	at androidx.compose.runtime.ComposerImpl.recomposeToGroupEnd(Composer.kt:2158)
	at androidx.compose.runtime.ComposerImpl.skipToGroupEnd(Composer.kt:2427)
	at dev.patrickgold.florisboard.app.ui.components.FlorisCardsKt.FlorisOutlinedBox-wK_Y8yA(FlorisCards.kt:297)
	at dev.patrickgold.florisboard.app.ui.settings.theme.ThemeEditorScreenKt$ThemeEditorScreen$1$5$4$invoke$$inlined$items$default$2.invoke(LazyDsl.kt:275)
	at dev.patrickgold.florisboard.app.ui.settings.theme.ThemeEditorScreenKt$ThemeEditorScreen$1$5$4$invoke$$inlined$items$default$2.invoke(LazyDsl.kt:100)
	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:135)
	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:34)
	at androidx.compose.foundation.lazy.list.LazyListScopeImpl$items$1$1.invoke(LazyListScopeImpl.kt:41)
	at androidx.compose.foundation.lazy.list.LazyListScopeImpl$items$1$1.invoke(LazyListScopeImpl.kt:41)
	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:107)
	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:34)
	at androidx.compose.runtime.CompositionLocalKt.CompositionLocalProvider(CompositionLocal.kt:228)
	at androidx.compose.runtime.saveable.SaveableStateHolderImpl.SaveableStateProvider(SaveableStateHolder.kt:84)
	at androidx.compose.foundation.lazy.layout.LazyLayoutItemContentFactory$CachedItemContent$content$1.invoke(LazyLayoutItemContentFactory.kt:103)
	at androidx.compose.foundation.lazy.layout.LazyLayoutItemContentFactory$CachedItemContent$content$1.invoke(LazyLayoutItemContentFactory.kt:94)
	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:107)
	at androidx.compose.runtime.internal.ComposableLambdaImpl.invoke(ComposableLambda.jvm.kt:34)
	at androidx.compose.runtime.RecomposeScopeImpl.compose(RecomposeScopeImpl.kt:140)
	at androidx.compose.runtime.ComposerImpl.recomposeToGroupEnd(Composer.kt:2158)
	at androidx.compose.runtime.ComposerImpl.skipCurrentGroup(Composer.kt:2404)
	at androidx.compose.runtime.ComposerImpl$doCompose$2$5.invoke(Composer.kt:2585)
	at androidx.compose.runtime.ComposerImpl$doCompose$2$5.invoke(Composer.kt:2571)
	at androidx.compose.runtime.SnapshotStateKt__DerivedStateKt.observeDerivedStateRecalculations(DerivedState.kt:247)
	at androidx.compose.runtime.SnapshotStateKt.observeDerivedStateRecalculations(Unknown Source:1)
	at androidx.compose.runtime.ComposerImpl.doCompose(Composer.kt:2571)
	at androidx.compose.runtime.ComposerImpl.recompose$runtime_release(Composer.kt:2547)
	at androidx.compose.runtime.CompositionImpl.recompose(Composition.kt:620)
	at androidx.compose.runtime.Recomposer.performRecompose(Recomposer.kt:786)
	at androidx.compose.runtime.Recomposer.access$performRecompose(Recomposer.kt:105)
	at androidx.compose.runtime.Recomposer$runRecomposeAndApplyChanges$2$2.invoke(Recomposer.kt:456)
	at androidx.compose.runtime.Recomposer$runRecomposeAndApplyChanges$2$2.invoke(Recomposer.kt:425)
	at androidx.compose.ui.platform.AndroidUiFrameClock$withFrameNanos$2$callback$1.doFrame(AndroidUiFrameClock.android.kt:34)
	at androidx.compose.ui.platform.AndroidUiDispatcher.performFrameDispatch(AndroidUiDispatcher.android.kt:109)
	at androidx.compose.ui.platform.AndroidUiDispatcher.access$performFrameDispatch(AndroidUiDispatcher.android.kt:41)
	at androidx.compose.ui.platform.AndroidUiDispatcher$dispatchCallback$1.doFrame(AndroidUiDispatcher.android.kt:69)
	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1035)
	at android.view.Choreographer.doCallbacks(Choreographer.java:845)
	at android.view.Choreographer.doFrame(Choreographer.java:775)
	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1022)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7840)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:550)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)

```
</details>

**Comment by patrickgold** [comment]
Thanks for your bug report!

You provided 2 crash logs at once. 

The first crash log (`java.lang.IllegalStateException: Check failed.`) is a very generic crash, I suspect it was either a bad character name used in a variable by you or some error. As there's no way to tell though which input caused the crash, I cannot fix it. In response to the lack of detail on that crash log I adjusted the crash reporter however to output more detailed info should it occur again.

The second one (`java.lang.IllegalArgumentException: The percent should be in the range of [0, 100]`) has been immediately reproduced and resolved by me, will not crash for percentage values over 100 anymore (by preventing you from even inputting it in the first place).

Both fixes will be released in 0.3.14-rc02 in the next days. Thanks for reporting!

**Comment by AnonWombat** [comment]
> Thanks for your bug report!
> 
> You provided 2 crash logs at once.
> 
> The first crash log (`java.lang.IllegalStateException: Check failed.`) is a very generic crash, I suspect it was either a bad character name used in a variable by you or some error. As there's no way to tell though which input caused the crash, I cannot fix it. In response to the lack of detail on that crash log I adjusted the crash reporter however to output more detailed info should it occur again.
> 
> The second one (`java.lang.IllegalArgumentException: The percent should be in the range of [0, 100]`) has been immediately reproduced and resolved by me, will not crash for percentage values over 100 anymore (by preventing you from even inputting it in the first place).
> 
> Both fixes will be released in 0.3.14-rc02 in the next days. Thanks for reporting!

Thanks!
